src/System.Management.Automation/CoreCLR/CorePsAssemblyLoadContext.cs

```diff
@@ -371,8 +371,8 @@ private bool TryFindInGAC(AssemblyName assemblyName, out string assemblyFilePath
                 {
                     if (string.IsNullOrEmpty(_gacPath64))
                     {
-                        // cache value of '_gacPath64' folder in member variable.
-                        _gacPath64 = $"{_winDir}{dirSeparator}Microsoft.NET{dirSeparator}assembly{dirSeparator}GAC_64";
+                       var gacName = RuntimeInformation.ProcessArchitecture == Architecture.Arm64 ? "GAC_Arm64" : "GAC_64";
+                        _gacPath64 = string.Create(CultureInfo.InvariantCulture, $"{_winDir}{dirSeparator}Microsoft.NET{dirSeparator}assembly{dirSeparator}{gacName}");
```

Comment:
Can we use `Path.Join(string, string, string, string)` (or `Path.Combine(s, s, s, s)`) to construct the path?
If not too much trouble, maybe also replace other `"${...}{dirSeparator}{...}..."` instances with `Path.Join` (or `Path.Combine`).

Reply:
Sure, was just using the current style, but makes sense